.github/workflows/test.yml

```diff
@@ -7,99 +7,51 @@ on:
-        with:
-          python-version: "3.x"
-
-      - name: Install dependencies
-        run: |
-          python -m pip install --upgrade pip
-          pip install twine wheel jupyter-packaging jupyterlab
-
-      - name: Build
-        run: |
-          python setup.py generate_source
-          python setup.py sdist bdist_wheel
-
-      - name: Build
-        run: |
-          cd js
-          npm pack
-
-      - name: Upload builds
-        uses: actions/upload-artifact@v3
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
         with:
-          name: ipyvuetify-dist-${{ github.run_number }}
-          path: |
-            ./dist
-            ./js/*.tgz
+          python-version: "3.10"
+      - name: install build
+        run: python -m pip install build
+      - name: build wheel
+        run: python -m build
+      - name: install ipyvuetify from the wheel
+        run: python -m pip install --find-links dist/ ipyvuetify
+      - name: test import
+        run: (mkdir test-install; cd test-install; python -c "from ipyvuetify import Btn")
 
   test:
-    needs: [build]
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9, "3.10", "3.11"]
+        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
     steps:
-      - uses: actions/checkout@v2
-
-      - uses: actions/download-artifact@v3
-        with:
-          name: ipyvuetify-dist-${{ github.run_number }}
-
-      - name: Install Python
-        uses: actions/setup-python@v2
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
-
-      - name: Install
-        run: pip install dist/*.whl
-
-      - name: Import
-        # do the import in a subdirectory, as after installation, files in de current directory are also imported
-        run: |
-          (mkdir test-install; cd test-install; python -c "from ipyvuetify import Btn")
+      - name: Install ipyvuetify
+        run: python -m pip install .
+      - name: test import
+        run: (mkdir test-install; cd test-install; python -c "from ipyvuetify import Btn")
 
   ui-test:
-    needs: [build]
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-
-      - uses: actions/download-artifact@v3
-        with:
-          name: ipyvuetify-dist-${{ github.run_number }}
-
-      - name: Install Python
-        uses: actions/setup-python@v2
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v3
         with:
-          python-version: 3.8
-
-      - name: Install vuetify and test deps
-        run: |
-          wheel=(dist/*.whl)
-          pip install ${wheel}[test] "jupyter_server<2"
-
-      - name: Install playwright browsers
+          python-version: "3.10"
+      - name: Install ipyvuetify
+        run: python -m pip install .[test]
```

Comment:
there is a "build" job dedicated to exactly that before test. Are you sure you want to rely on the produced wheel (preventing the parralel execution) ?

Reply:
How would it execute parallel? It will not speed up anything I think? Instead, every job would need to build its own wheel, only wasting resources. In my experience, the more jobs you have, the slower CI runs (my guess is they all share a fixed amount of CPU resource).